wiki/proposals/PIP.md

```diff
@@ -0,0 +1,146 @@
+
+4. **REJECTED**: The proposal has not been accepted by the Pulsar project.
+
+5. **IMPLEMENTED**: The implementation of the proposed changes have been
+   completed and everything has been merged.
+
+5. **RELEASED**: The proposed changes have been included in an official
+   Apache Pulsar release.
+
+The process works in the following way:
+
+1. The author(s) of the proposal will create a GitHub issue ticket choosing the
+   template for PIP proposals.
+2. The author(s) will send a note to the dev@pulsar.apache.org mailing list
+   to start the discussion, using subject prefix `[PIP] xxx`.
+3. Based on the discussion and feedback, some changes might be applied by
+   authors to the text of the proposal.
+4. Once some consensus is reached, there will be a vote to formally approve
+   the proposal.
+   The vote will be held on the dev@pulsar.apache.org mailing list. Everyone
+   is welcome to vote on the proposal, though it will considered to be binding
+   only the vote of PMC members.
+   I would be required to have a lazy majority of at least 3 binding +1s votes.
+   The vote should stay open for at least 48 hours.
+5. When the vote is closed, if the outcome is positive, the state of the
+   proposal is updated and the Pull Requests associated with this proposal can
+   start to get merged into the master branch.
+
+All the Pull Requests that are created, should always reference the
+PIP-XXX in the
+commit log message and the PR title.
+
+## Labels of a PIP
+
+In addition to its current state, the GitHub issue for the PIP will also be
+tagged with other labels.
+
+Some examples:
+* Execution status: In progress, Completed, Need Help, ...
+* Targeted Pulsar release: 2.9, 2.10, ...
+
+
+## Template for a PIP design doc
+
+```
+## Motivation
+
+Explain why this change is needed, what benefits it would bring to Apache Pulsar
+and what problem it's trying to solve.
+
+## Goal
+
+Define the scope of this proposal. Given the motivation stated above, what are
+the problems that this proposal is addressing and what other items will be
+considering out of scope, perhaps to be left to a different PIP.
+
+## API Changes
+
+Illustrate all the proposed changes to the API or wire protocol, with examples
+of all the newly added classes/methods, including Javadoc.
+
+## Implementation
+
+This should be a detailed description of all the changes that are
+expected to be made. It should be detailed enough that any developer that is
+familiar with Pulsar internals would be able to understand all the parts of the
+code changes for this proposal.
+
+This should also serve as documentation for any person that is trying to
+understand or debug the behavior of a certain feature.
+
+
+## Reject Alternatives
+
+If there are alternatives that were already considered by the authors or,
+after the discussion, by the community, and were rejected, please list them
+here along with the reason why they were rejected.
+
+```
```

Comment:
Duplicate to the PIP-template. We can remove this content.

If you're not in a hurry. Let's review the content. Or if you want to push this migration first (since the content is already there for a long time) and improve the content later, we can do it as a follow-up.